### ISSUE TITLE
integral.py failures on both vanilla and s-o-g

### ISSUE THREAD

**Issue by strogdon** [cschwan/sage-on-gentoo#721]
```
sage -t --random-seed=118672341557027818714214832529596263930 /usr/lib/python3.10/site-packages/sage/symbolic/integration/integral.py
**********************************************************************
File "/usr/lib/python3.10/site-packages/sage/symbolic/integration/integral.py", line 203, in sage.symbolic.integration.integral.DefiniteIntegral.__init__
Failed example:
    integral(ex, x, 0, 2, algorithm='giac')
Expected:
    3/2
Got:
    undef
**********************************************************************
File "/usr/lib/python3.10/site-packages/sage/symbolic/integration/integral.py", line 205, in sage.symbolic.integration.integral.DefiniteIntegral.__init__
Failed example:
    integral(1/max_symbolic(x, 1)**2, x, 0, oo, algorithm='giac')
Expected:
    2
Got:
    +Infinity
**********************************************************************
1 item had failures:
   2 of   6 in sage.symbolic.integration.integral.DefiniteIntegral.__init__
    [238 tests, 2 failures, 15.04 s]
----------------------------------------------------------------------
sage -t --random-seed=118672341557027818714214832529596263930 /usr/lib/python3.10/site-packages/sage/symbolic/integration/integral.py  # 2 doctests failed
```
Is this `giac` related? System `giac` is used.

**Comment by strogdon** [comment]
System installed `giac` is `giac-1.7.0.13-r1`.

**Comment by kiwifb** [comment]
https://trac.sagemath.org/ticket/32354 minimum version of giac needs bumping.

**Comment by kiwifb** [comment]
Fixed by 08858af478dfb4ca5058f654ad90e00f07022937 and 42537a24dba28405e85f887c3ce0253dde1cd80d.

**Comment by strogdon** [comment]
Vanilla's configure may need some revision. From `config.log`
```
configure:26431: checking for giac >= 1.5.0, <= 1.9.999
configure:26551: result: /usr/bin/giac
configure:26557: checking for giac/giac.h
configure:26557: g++ -std=gnu++11 -c    -Dlinux   -I/usr/include conftest.cpp >&5
configure:26557: $? = 0
configure:26557: result: yes
```
and
```
configure:26590: g++ -std=gnu++11 -o conftest    -Dlinux   -I/usr/include -Wl,-O1 -Wl,--as-needed conftest.cpp -lgiac  -lecm -lpari -lcurl -lcliquer -lcddgmp -lbz2 -larb -lflint -lmpfr -lglpk -lgmp -lm  -lntl >&5
configure:26590: $? = 0
configure:26610: result: -lgiac
configure:26638: will use system package and not install SPKG giac
```
